hyperspace/testsuite/src/lib.rs

```diff
@@ -594,3 +594,32 @@ where
 	send_packet_and_assert_timeout_on_channel_close(chain_a, chain_b, channel_id).await;
 	handle.abort()
 }
+
+pub async fn client_synchronization_test<A, B>(chain_a: &A, chain_b: &B)
+where
+	A: TestProvider,
+	A::FinalityEvent: Send + Sync,
+	A::Error: From<B::Error>,
+	B: TestProvider,
+	B::FinalityEvent: Send + Sync,
+	B::Error: From<A::Error>,
+{
+	tokio::time::sleep(Duration::from_secs(60 * 20)).await;
+	let messages_to_b = chain_a.fetch_mandatory_updates(chain_b).await.unwrap();
+	let messages_to_a = chain_b.fetch_mandatory_updates(chain_a).await.unwrap();
+	log::info!(target: "hyperspace",
+		"Syncing Client on {} {:?}",
+		chain_a.name(),
+		messages_to_a.clone().into_iter().map(|msg| msg.type_url).collect::<Vec<_>>()
+	);
+	log::info!(target: "hyperspace",
+		"Syncing Client on {} {:?}",
+		chain_b.name(),
+		messages_to_b.clone().into_iter().map(|msg| msg.type_url).collect::<Vec<_>>()
+	);
+	chain_a.submit(messages_to_a).await.unwrap();
+	chain_b.submit(messages_to_b).await.unwrap();
+
```

Comment:
shouldn't we check that the client is synced?

Reply:
Doing that in the refactored code